### PR TITLE
Report Version when Testing Repositories

### PR DESCRIPTION
**Pull request by aerique**
### Short description

This adds a minor feature to the repo test script where it reports the versions of the packages that were tested.

Ubuntu Bionic has been removed for auth-48 and dnsdist-18.

The script's version has finally been updated (as it should on every change).

### Example Output

```
$ ./generate-repo-files.py --test rec-48
=== testing rec-48 ===
Building Docker image using Dockerfile.rec-48.centos-7...
Running Docker container tagged rec-48.centos-7...
Building Docker image using Dockerfile.rec-48.debian-bullseye...
Running Docker container tagged rec-48.debian-bullseye...
Building Docker image using Dockerfile.rec-48.debian-buster...
Running Docker container tagged rec-48.debian-buster...
Building Docker image using Dockerfile.rec-48.el-8...
Running Docker container tagged rec-48.el-8...
Building Docker image using Dockerfile.rec-48.el-9...
Running Docker container tagged rec-48.el-9...
Building Docker image using Dockerfile.rec-48.ubuntu-bionic...
Running Docker container tagged rec-48.ubuntu-bionic...
Building Docker image using Dockerfile.rec-48.ubuntu-focal...
Running Docker container tagged rec-48.ubuntu-focal...
Building Docker image using Dockerfile.rec-48.ubuntu-jammy...
Running Docker container tagged rec-48.ubuntu-jammy...
Test done.
- returned versions:
    - rec-48.centos-7: 4.8.3
    - rec-48.debian-bullseye: 4.8.3
    - rec-48.debian-buster: 4.8.3
    - rec-48.el-8: 4.8.3
    - rec-48.el-9: 4.8.3
    - rec-48.ubuntu-bionic: 4.8.3
    - rec-48.ubuntu-focal: 4.8.3
    - rec-48.ubuntu-jammy: 4.8.3
```

### Checklist

I have:
- [ ] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by aerique** [comment]
It's just proper program etiquette I think.

(But forgetting to update it every time is not, so yeah... maybe we should remove it :shrug: )

File: build-scripts/docker/repo-test/generate-repo-files.py [hunk 1]
```diff
@@ -166,6 +165,12 @@ def write_release_files (release):
         write_dockerfile('debian', 'bullseye', release)
         write_list_file('debian', 'bullseye', release)
 
+    if release in ['auth-46', 'auth-47', 'auth-master',
+                   'rec-45', 'rec-46', 'rec-47', 'rec-48', 'rec-master',
```
[the remark this answers is not in the document]
Comment (omoerbeek) [reply build-scripts/docker/repo-test/generate-repo-files.py:169]: Not actually doing a review as my pythojn knowledge is *very* limited.

**Comment by Habbie** [comment]
> (But forgetting to update it every time is not, so yeah... maybe we should remove it shrug )

I vote for this :)

File: build-scripts/docker/repo-test/generate-repo-files.py [hunk 2]
```diff
@@ -205,13 +210,21 @@ def run (tag):
         capture_run_output = not(g_verbose)
     print('Running Docker container tagged {}...'.format(tag))
     cp = subprocess.run(['docker', 'run', tag],
-                        capture_output=capture_run_output)
+                        stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    version = re.search('(PowerDNS Authoritative Server|PowerDNS Recursor|' +
+                         'dnsdist) (\d+\.\d+\.\d+(-\w+)?)',
```
Comment (Habbie) [inline build-scripts/docker/repo-test/generate-repo-files.py:215]: I wonder if this expression is wide enough, but we'll find out. It looks fine for tagged releases at least :)

**Review by Habbie — APPROVED** [review]
approved. Version number can go, now or later.

**Pull request merged**